framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -584,6 +587,26 @@ private void printStoredMessages(CompilationUnitTree unit) {
         if (messageStore != null) {
             for (CheckerMessage msg : messageStore) {
                 super.printOrStoreMessage(msg.kind, msg.message, msg.source, unit);
+                this.printStackTrace(msg.trace);
+            }
+        }
+    }
+
+    /**
+     * Prints given stack trace of runtime exception if -AdumpOnErrors flag is passed as compilation
+     * argument
+     *
+     * @param trace stack trace when the checker encounters an error
+     */
+    private void printStackTrace(StackTraceElement[] trace) {
+        boolean dumpOnErrors =
+                (processingEnv != null
+                        && processingEnv.getOptions() != null
+                        && processingEnv.getOptions().containsKey("dumpOnErrors"));
+        if (dumpOnErrors) {
+            System.err.println(new RuntimeException().toString());
+            for (StackTraceElement elem : trace) {
+                System.err.println("\tat " + elem);
```

Comment:
Don't use `System.err` for any compiler output, always use the proper messager.
I would suggest using https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L1058 with diagnostic kind `NOTE`.
As you also have the diagnostic source, you could also use that location for the warning. How does `doe` output the message?

Reply:
I have used the given function to print the stack trace. I observed the -doe outputs using a printWriter in the Log.java file.